src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Model/AzureSqlManagedInstanceModel.cs

```diff
@@ -0,0 +1,96 @@
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Security;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Model
+{
+    /// <summary>
+    /// Represents the core properties of an Managed instance
+    /// </summary>
+    public class AzureSqlManagedInstanceModel
+    {
+        /// <summary>
+        /// Gets or sets the location the managed instance is in
+        /// </summary>
+        public string Location { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource ID.
+        /// </summary>
+        public string Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource group name.
+        /// </summary>
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the managed instance
+        /// </summary>
+        public string ManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tags associated with the managed instance.
+        /// </summary>
+        public Dictionary<string, string> Tags { get; set; }
+
+        /// <summary>
+        /// Gets or sets the identity of the managed instance.
+        /// </summary>
+        public Management.Sql.Models.ResourceIdentity Identity { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Sku of the managed instance
+        /// </summary>
+        public Management.Internal.Resources.Models.Sku Sku { get; set; }
+
+        /// <summary>
+        /// Gets or sets the fully qualified domain name of the managed instance
+        /// </summary>
+        public string FullyQualifiedDomainName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the sql login credentials for the admin
+        /// </summary>
+        public string AdministratorLogin { get; set; }
+
+        /// <summary>
+        /// Gets or sets the password for the sql admin
+        /// </summary>
+        public SecureString AdministratorPassword { get; set; }
+
+        /// <summary>
+        /// Gets or sets subnet resource ID for the managed instance.
+        /// </summary>
+        public string SubnetId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the license type. Possible values are
+        /// 'LicenseIncluded' and 'BasePrice'.
+        /// </summary>
+        public string LicenseType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the number of VCores.
+        /// </summary>
+        public int? VCores { get; set; }
```

Comment:
New-MI has `-Vcore`, not `-Vcores`. should this be consistent?

Reply:
Yes, I used "Vcore" in New-MI parameters to match the SQL db experience (and cmdlet param shoud be singular), but our API uses "VCores". Not sure should I use "VCores" as cmdlet parameter?